switchboard/sql/SCHEMAv2.sql

```diff
@@ -0,0 +1,249 @@
+    email         text                      not null unique,
+    password_hash text                      not null,
+
+    user_type     tml_switchboard.user_type not null,
+    locked        bool                      not null default false
+);
+
+-- API tokens are rather vitally important. They are in fact the only 'mark' of authorization: only with a token can a
+-- request from a client be authorized.
+-- API tokens can either have their own permissions, or inherit from the user that owns the token.
+-- TODO: it is not clear whether an inheriting token can have its own privileges: however, it seems unlikely, since it
+--       seems a logical invariant that privilege should be monotonic: a token's privileges should be a weakening of
+--       those of the user that owns it.
+-- API tokens have both an expiration system and a cancellation mechanism, which can be used to render a token invalid
+-- and void before its natural expiration.
+create type tml_switchboard.api_token_cancellation as
+(
+    canceled_at         timestamp with time zone,
+    cancellation_reason text
+);
+create table tml_switchboard.api_tokens
+(
+    token_id                  uuid                     not null primary key,
+    token                     bytea                    not null unique,
+    user_id                   uuid                     not null references tml_switchboard.users (user_id) on delete cascade,
+    inherits_user_permissions bool                     not null,
+    canceled                  tml_switchboard.api_token_cancellation,
+    created_at                timestamp with time zone not null,
+    expires_at                timestamp with time zone not null,
+
+    check (octet_length(token) = 128)
+);
+
+-- Privileges are assignments of permissions to subjects (users and tokens). Note that is not necessary for a token to
+-- explicitly be assigned privileges (the inheritance clause described for API tokens above).
+create table tml_switchboard.user_privileges
+(
+    user_id    uuid not null references tml_switchboard.users (user_id) on delete cascade,
+    permission text not null,
+
+    primary key (user_id, permission)
+);
+create table tml_switchboard.api_token_privileges
+(
+    token_id   uuid not null references tml_switchboard.api_tokens (token_id) on delete cascade,
+    permission text not null,
+
+    primary key (token_id, permission)
+);
+
+-- All supervisors must be registered with the switchboard database: any supervisor not registered will be turned away
+-- at the gate, so to speak.
+-- The name is at the moment superfluous and nowhere referenced, though that may change in the future.
+-- The auth_token field is quite the opposite-a supervisor attempting to connect to the switchboard must provide proof
+-- of its identity, and that is precisely what the auth_token provides: a 256-byte random string uniquely authenticating
+-- the supervisor.
+-- TODO: The tags remain ill-specified.
```

Comment:
Indeed, we should address this in a follow-up change. For now, a simple "supervisor offers tags", "job requires tags" match strategy should suffice, but may not in the future.